### PR TITLE
Add basic client tests.

### PR DESCRIPTION
**Pull request by jakobr-google**

### CODE REVIEW

File: test/src/client_utils.dart [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+class MockChannel extends Mock implements ClientChannel {}
+
+typedef ServerMessageHandler = void Function(StreamMessage message);
+
+class TestClient {
+  final ClientChannel _channel;
+
+  static final _$unary = new ClientMethod<int, int>(
+      '/Test/Unary',
+      (int value) => new List.filled(value, 0),
```
Comment (johnniwinther) [inline test/src/client_utils.dart]: Extract these as `mockEncode` and `mockDecode` functions to use in the client methods and in the request handlers in the tests.

File: test/client_test.dart [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+  tearDown(() {
+    harness.tearDown();
+  });
+
+  test('Unary calls work end-to-end', () async {
+    void _handleRequest(StreamMessage message) {
+      expect(message, new isInstanceOf<DataStreamMessage>());
+      expect(message.endStream, false);
+      final data = new GrpcHttpDecoder().convert(message) as GrpcData;
+      expect(data.data.length, 17);
```
Comment (johnniwinther) [inline test/client_test.dart]: Extract these numbers to clearly show that they magic values.

Comment (jakobr-google) [inline test/client_test.dart]: Done.

File: test/src/client_utils.dart (hunk 1 above)
Comment (jakobr-google) [inline test/src/client_utils.dart]: Done.

**Pull request merged**